.gitattributes

```diff
@@ -0,0 +1,288 @@
+# Common settings that generally should always be used with your language specific settings
```

Comment:
I haven't seen many repos include _all_ these settings -`.gitattributes`, `.gitignore`, .. -- are they necessary in this PR?

Reply:
I agree, we should only have necessary files, maybe @Richienb can elaborate a bit?